.cargo/config.toml

```
@@ -1,3 +1,17 @@
 [alias]
 xtask = "run --package xtask --"
 integration-test = "test --features integration --profile integration --workspace --test integration"
+
+# By default, on musl linux, statically linked binaries are produced.
+# But such binaries do not support loading libraries using `dlopen()` ("libloading" crate),
+# which is required for loading treesitter grammars.
+[target.arm-unknown-linux-musleabihf]
+rustflags = ["-C", "target-feature=-crt-static"]
+[target.armv7-unknown-linux-musleabihf]
+rustflags = ["-C", "target-feature=-crt-static"]
+[target.aarch64-unknown-linux-musl]
+rustflags = ["-C", "target-feature=-crt-static"]
+[target.i686-unknown-linux-musl]
+rustflags = ["-C", "target-feature=-crt-static"]
+[target.x86_64-unknown-linux-musl]
+rustflags = ["-C", "target-feature=-crt-static"]
```

You can [match by `target_env`](https://doc.rust-lang.org/cargo/reference/config.html#target) which should cover all of these? (_the [`musleabihf` variant will also match as `musl`](https://doc.rust-lang.org/reference/conditional-compilation.html#target_env)_)

You can also provide a string instead of an array for `rustflags`, AFAIK there is no practical difference ([rustflags from other sources are mutually exclusive](https://doc.rust-lang.org/cargo/reference/config.html#buildrustflags), no merging_).

```suggestion
[target.'cfg(target_env = "musl")']
rustflags = "-C target-feature=-crt-static"
```

Do note the [caveat of building with/without an explicit `--target`](https://doc.rust-lang.org/cargo/reference/config.html#buildrustflags). A [related issue](https://github.com/rust-lang/rust/issues/78210#issuecomment-714600426) where a build failed (_with `+crt-static` on a host with `-gnu` as the default implicit target_).